agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/quickpulse/QuickPulsePingSender.java

```diff
@@ -91,7 +91,7 @@ public QuickPulsePingSender(
 
   public QuickPulseHeaderInfo ping(String redirectedEndpoint) {
     String instrumentationKey = getInstrumentationKey();
-    if (instrumentationKey == null && "java".equals(System.getenv("FUNCTIONS_WORKER_RUNTIME"))) {
+    if (Strings.isNullOrEmpty(instrumentationKey)) {
```

Comment:
just curios, is there a reason the function runtime check was removed?

Reply:
it is no longer applied to function only. azure spring cloud with empty ikey, no live metrics.  that's why we drop that check.